torch/_dynamo/utils.py

```diff
@@ -913,6 +919,61 @@ def enum_repr(value, local):
     return local_name
 
 
+def is_hashable_python_var(x):
+    from .allowed_functions import is_builtin_callable
+    from .variables import ConstantVariable
+
+    return (
+        ConstantVariable.is_literal(x)
+        or isinstance(x, torch.Tensor)
+        or is_builtin_callable(x)
+    )
+
+
+def is_hashable(x):
+    from .variables import (
+        BuiltinVariable,
+        ConstantVariable,
+        EnumVariable,
+        TensorVariable,
+    )
+
+    return isinstance(
+        x,
+        (
+            BuiltinVariable,
+            ConstantVariable,
+            EnumVariable,
+            TensorVariable,
+        ),
+    )
+
+
+class HashableTracker:
+    def __init__(self, vt):
+        assert is_hashable(vt), type(vt)
+        self.vt = vt
+
+    @property
+    def underlying_value(self):
+        from .variables import TensorVariable
+
+        if isinstance(self.vt, TensorVariable):
+            x = self.vt.as_proxy().node
+        else:
+            x = self.vt.as_python_constant()
```

Comment:
Hmm, will this work for SymNode?

Reply:
You'd probably have to hash the backing value of the symnode, right?

Reply:
Not really. For a symnode we need to perform id check. If that fails, we would need to either specialise the value or do something like what we do in the `eq` check, and perhaps specialise later

Reply:
Oh I see. ID check of the symnode itself would just be an escape hatch to avoid specializing. 

Because python literals will hash/id to same value. Two separate symnode objects can represent the same specialized value.